### PR TITLE
Winston loggly

### PR DESCRIPTION
**Pull request by Kourchenko**
* Added Loggly to Winston.
* Fixed helper function 'getEnvironmentVariables' to export simply.

### CODE REVIEW

File: src/common/logging.ts [hunk 1]
```diff
@@ -1,22 +1,24 @@
...
 /**
  * Winston Log Object
  * Send logs to Console if message level <= 'error'.
  * Send logs to Sentry if message level is at least 'info'.
  */
+const envVariables = getEnvironmentVariables();
 const logging =
   createLogger({
-    level: getEnv() === 'production' ? 'info' : 'debug',
+    level: envVariables.env === 'production' ? 'error' : 'debug',
```
Comment (ryanrolds) [inline src/common/logging.ts]: Let's have `production` set this to `level`. Down the road, we may change it to `warn`.

File: src/common/config.ts [hunk 2]
```diff
@@ -1,6 +1,12 @@
 
-function getEnv() {
-  return process.env['NODE_ENV'];
+function getEnvironmentVariables() {
+  const env:string | undefined = process.env['NODE_ENV'];
```
Comment (ryanrolds) [inline src/common/config.ts]: If not set can we default to 'local' or throw an error? It should be set (with a default) or the service shouldn't start (error).

File: src/common/logging.ts [hunk 3]
```diff
@@ -29,11 +31,28 @@ const logging =
...
         consoleWarnLevels: ['warn', 'debug', 'info'],
       }),
-      new Sentry({
-        dsn: 'https://34760911653f40a18ba07c39a6a37002@sentry.io/1488146',
-        level: 'info',
-      }),
     ],
   });
 
+if (envVariables.logglyToken !== null &&
```
Comment (ryanrolds) [inline src/common/logging.ts]: `if (!envVariables.logglyToken) { ... }` is the same as making sure it's not null, undefined, or an empty string.

File: src/common/logging.ts [hunk 4]
```diff
@@ -29,11 +31,28 @@ const logging =
...
+    envVariables.logglyToken !== undefined) {
+  winston.add(new Loggly({
+    level: 'error',
```
Comment (ryanrolds) [inline src/common/logging.ts]: This should be the same log level as the other transport.

File: src/common/logging.ts [hunk 5]
```diff
@@ -1,22 +1,24 @@
...
  * Winston Log Object
  * Send logs to Console if message level <= 'error'.
  * Send logs to Sentry if message level is at least 'info'.
  */
+const envVariables = getEnvironmentVariables();
 const logging =
   createLogger({
-    level: getEnv() === 'production' ? 'info' : 'debug',
+    level: envVariables.env === 'production' ? 'error' : 'debug',
     defaultMeta: {service: 'speedupamerica-frontend'},
```
Comment (ryanrolds) [inline src/common/logging.ts]: Please include hostname. 

When I look at the logout:
```
info: Starting main.ts {"service":"speedupamerica-frontend","timestamp":"2019-07-17T01:50:00.354Z"}
info: Stopping main.ts {"service":"speedupamerica-frontend","timestamp":"2019-07-17T01:50:00.356Z"}
```
Can we get the message in the JSON as well as the level?

**Comment by ryanrolds** [comment]
One last thing, make sure to check whats gets send to Loggly by setting the LOGGLY_TOKEN in your local env for a little bit.

File: src/common/config.ts (hunk 2 above)
Comment (Kourchenko) [reply src/common/config.ts]: This is a good idea. Something like
`const env = process.env['NODE_ENV'] || 'local';`

File: src/common/logging.ts (hunk 5 above)
Comment (Kourchenko) [reply src/common/logging.ts]: Added `message `and `level`.

File: src/common/config.ts [hunk 6]
```diff
@@ -1,6 +1,16 @@
 
-function getEnv() {
-  return process.env['NODE_ENV'];
+function environmentVariables() {
+  const NODE_ENV: string = process.env['NODE_ENV'] || 'local';
+  const DB_HOSTNAME: string = process.env['NODE_ENV'] || 'localhost';
```
Comment (ryanrolds) [inline src/common/config.ts]: Not the DB hostname, but [os.hostname()](https://nodejs.org/api/os.html#os_os_hostname).

Comment (Kourchenko) [reply src/common/config.ts]: Oh. I was wondering about whether the OS Hostname or the DB hostname. Will fix!

**Pull request merged**